server/models/Notification.ts

```diff
@@ -232,41 +232,58 @@ class Notification extends Model<
    * @param notification Notification for which the past notifications are fetched - used for determining the properties that form a thread.
    * @returns An array of email message ids that form a thread.
    */
-  static async references(
+  public static async emailReferences(
     notification: Notification
   ): Promise<string[] | undefined> {
     if (!isEmailThreadSupportedNotification(notification.event)) {
       return;
     }
 
-    const prevNotifications = await this.unscoped().findAll({
-      attributes: ["id"],
-      where: {
-        id: {
-          [Op.ne]: notification.id,
+    const prevNotifications: Notification[] = [];
+    const limit = 100;
+    let offset = 0,
+      hasMore = true;
+
+    while (hasMore) {
+      const loadedNotifications = await this.unscoped().findAll({
+        attributes: ["id"],
+        where: {
+          id: {
+            [Op.ne]: notification.id,
+          },
+          event: notification.event,
+          documentId: notification.documentId,
+          userId: notification.userId,
         },
-        event: notification.event,
-        documentId: notification.documentId,
-        userId: notification.userId,
-      },
-      order: [["createdAt", "ASC"]],
-    });
+        order: [["createdAt", "ASC"]],
+        offset,
+        limit,
+      });
+
+      if (!loadedNotifications.length) {
+        hasMore = false;
+      }
+
+      prevNotifications.push(...loadedNotifications);
+      offset += limit;
+    }
```

Comment:
We have `findAllInBatches` method if that's what you're looking for here.

Reply:
Ah great!! I wasn't aware of this, thanks 😄